.github/workflows/build-push.yaml

```diff
@@ -0,0 +1,61 @@
+name: Build and Push
+
+on:
+  workflow_call:
+    inputs:
+      dir:
+        description: Directory to build and push
+        required: true
+        default: ''
+        type: string
+
+jobs:
+  build-push:
+    name: Build and Push
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v2
+      with:
+        platforms: 'arm64'
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Docker meta
+      id: docker_meta
+      uses: docker/metadata-action@v4
+      with:
+        images: ghcr.io/${{ github.repository }}
+        flavor: |
+          suffix=-${{ inputs.dir }},onlatest=true
+        tags: |
+          type=sha
+          type=edge
+          type=ref,event=pr
+          type=semver,pattern={{raw}}
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ github.token }}
+
+    - name: Build and push
+      id: docker_build
+      uses: docker/build-push-action@v3
+      with:
+        context: ./${{ inputs.dir }}
+        file: ./${{ inputs.dir }}/Dockerfile
+        push: true
+        tags: ${{ steps.docker_meta.outputs.tags }}
+        labels: ${{ steps.docker_meta.outputs.labels }}
```

Comment:
Do we want to add caching? (see e.g. https://github.com/thestormforge/optimize-trials/pull/7/files#diff-ee31744c16c09322dc783a319da3c759dd38bf18f444b8f5f1ed0fb4ef49e65cR33-R34).